client_packages/chat/js/chat.js

```diff
@@ -0,0 +1,172 @@
+const BUTTON_T = 84,
+    BUTTON_ENTER = 13,
+    BUTTON_ESCAPE = 27,
+    BUTTON_ARROW_UP = 38,
+    BUTTON_ARROW_DOWN = 40;
+
+let chat = {
+    size: 0,
+    container: null,
+    input: null,
+    enabled: false,
+    active: true,
+    timer: null,
+    previous: new Array(),
+    messageNumber: 0,
+    hide_chat: 15*1000 // 15 - seconds
+};
+var langRegex = {
+    "EN": /^[a-zA-Z]+$/,
+    "RU": /[а-яА-ЯёЁ]/,
+    "UA": /[ієїґ\']+/ig
+};
+function enableChatInput(enable)
```

Comment:
We can refactor this func to something like that (but im not sure that it is better, @vitalik1188 check it please

```
function enableChatInput(enable) {
    if (chat.active === false && enable === true) {
        return;
    }
    if (enable === (chat.input !== null)) {
        chat.input.fadeOut('fast', function () {
            chat.input.remove();
            chat.input = null;
            mp.trigger("changeChatState", false);
        });
        return;
    }
    mp.invoke("focus", enable);
    if (enable) {
        $("#chat").css("opacity", 1);
        chat.input = $("#chat").append('<div><input id="chat_msg" type="text" /><div id="chat_lang">EN</div></div>').children(":last");
        chat.input.children("input").focus();

        chat.input.children("input").on('keypress', function (e) {
            Object.entries(langRegex).forEach(([key, value]) => {
                if (value.test(e.key)) {
                    $("#chat_lang").text(key);
                }
            });
        });
        mp.trigger("changeChatState", true);
    }
}
```